h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,224 @@
+            final SharedTreeGraph g = treeBackedModel.convert(treeToPrint, null, options);
+            printPng(g);
+        }
+        else {
+            System.out.println("ERROR: Unknown MOJO type");
+            System.exit(1);
+        }
+    }
+
+    private void printPng(SharedTreeGraph trees) throws IOException, ImportException {
+        Path outputDirectoryPath = Paths.get(outputFileName);
+        int numberOfTrees = trees.subgraphArray.size();
+        if (Files.exists(outputDirectoryPath)) {
+            Files.walk(outputDirectoryPath)
+                    .sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .forEach(File::delete);
+        }
+        if (numberOfTrees > 1) {
+            Files.createDirectory(outputDirectoryPath);
+        }
+        for (SharedTreeSubgraph tree : trees.subgraphArray) {
+            Path dotSourceFilePath = Files.createTempFile("", tmpOutputFileName);
+            try (FileOutputStream fosTemp = new FileOutputStream(dotSourceFilePath.toFile()); PrintStream osTemp = new PrintStream(fosTemp)) {
+                tree.printDot(osTemp, maxLevelsToPrintPerEdge, detail, optionalTitle, pTreeOptions, true);
+                generateOutputPng(dotSourceFilePath, numberOfTrees, tree.name);
+            }
+            Files.delete(dotSourceFilePath);
+        }
+
+    }
+
+    private void generateOutputPng(Path dotSourceFilePath, int numberOfTrees, String treeName) throws ImportException, IOException {
+        LabeledVertexProvider vertexProvider = new LabeledVertexProvider();
+        LabeledEdgesProvider edgesProvider = new LabeledEdgesProvider();
+        ComponentUpdater componentUpdater = new ComponentUpdater();
+        DOTImporter<LabeledVertex, LabeledEdge> importer = new DOTImporter<>(vertexProvider, edgesProvider, componentUpdater);
+        DirectedMultigraph<LabeledVertex, LabeledEdge> result = new DirectedMultigraph<>(LabeledEdge.class);
+        try (FileInputStream is = new FileInputStream(dotSourceFilePath.toFile())) {
+            importer.importGraph(result, new InputStreamReader(is));
+            JGraphXAdapter<LabeledVertex, LabeledEdge> graphAdapter = new JGraphXAdapter<LabeledVertex, LabeledEdge>(result);
+            mxIGraphLayout treeLayout = new mxCompactTreeLayout(graphAdapter, true);
+            mxIGraphLayout nonOverlappingEdgesLayout = new mxEdgeLabelLayout(graphAdapter);
+            treeLayout.execute(graphAdapter.getDefaultParent());
+            nonOverlappingEdgesLayout.execute(graphAdapter.getDefaultParent());
+            BufferedImage image = mxCellRenderer.createBufferedImage(graphAdapter, null, 2, Color.WHITE, true, null);
+            if (outputFileName != null) {
+                ImageIO.write(image, "PNG", new File(getPngName(numberOfTrees, treeName)));
+            } else {
+                ImageIO.write(image, "PNG", System.out);
```

Comment:
we probably should validate that png output with multiple trees also sets outputFile, writing multiple PNGs to stdout is not very useful

Reply:
fixed

Reply:
I dont see it